### PR TITLE
Security audit fixes.

### PR DESCRIPTION
**Pull request by this-pama**
- [x] Add helmet to secure Express apps by setting HTTP response headers. I have defined a `csp_links` object in the config file. It contains the list of all trusted domain names needed to load styles, javascript and images.
- [x] Remove overly descriptive messages for login errors.
- [x] Check user rights before allowing contributor profile updates.
- [x] Add middleware to check if request inputs, queries, or parameters contain HTML or script to avoid XSS injections. This helps sanitise user inputs and ensures that no unwanted HTML or javascript tags are in any user-supplied inputs to avoid cross-site scripting (XSS) attacks.
- [x] Default page language to 'en' if language is null. Fix for #135 
- [x] Add rate limiter to avoid DDoS attacks. This PR adds a rate limiter to the login endpoint. If there are 10 failed login attempts from the same IP address within 1 hour, the system blocks the IP address from login for 3 hours. This helps protect from DDoS and brute force attacks as recommended by [ExpressJs](https://expressjs.com/en/advanced/best-practice-security.html)

### CODE REVIEW

File: views/partials/head.ejs [hunk 1]
```diff
@@ -109,7 +109,7 @@
 
 	}
 
-	function rewriteUrl(lang, reload = false){
+	function rewriteUrl(lang = 'en', reload = false){
```
Comment (JosuaKrause) [inline views/partials/head.ejs:112]: how about
```suggestion
	function rewriteUrl(lang, reload = false){
	    if (!lang) {
	        lang = 'en';
	    }
```
this way it also works if someone explicitly passes a null (ie., `rewriteUrl(null, false)`

File: routes/login/check.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 	const token = req.body.token || req.query.token || req.headers['x-access-token']
 	const { uuid } = req.session || {}
 	
 	const cookies = parseCookies(req)
 	let sid = cookies[`${app_suite}-session`]
 	if (sid) sid = sid.split(':')[1].split('.')[0]
-	console.log(cookies)
-	console.log(sid)
-
+	// console.log(cookies)
```
Comment (JosuaKrause) [inline routes/login/check.js:14]: let's remove them altogether, no?

File: routes/helpers/rights/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const { DB } = include('config/')
+
+module.exports = async (_kwargs) => {
+	const conn = _kwargs.connection || DB.general
+	const { uuid } = _kwargs
+
+	const right = await conn.oneOrNone(`
+		SELECT COALESCE(rights, 0) AS rights FROM users
```
Comment (myjyby) [inline routes/helpers/rights/index.js:8]: cast to int > COALESCE (rights, 0)::INT

File: routes/helpers/rights/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+		WHERE uuid = $1
+	;`, [ uuid ])
```
Comment (myjyby) [inline routes/helpers/rights/index.js:10]: ;`, [ uuid ], d => d.rights)

File: routes/helpers/rights/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+	const rights = right?.rights ?? 0;
```
Comment (myjyby) [inline routes/helpers/rights/index.js:12]: then this is not needed and you can just send "right" (the first variable))

**Review by myjyby — APPROVED** [review]
lgtm

File: views/partials/head.ejs (hunk 1 above)
Comment (JosuaKrause) [reply views/partials/head.ejs:112]: okay, I'll do it on my side then. The change you're doing here doesn't change the behavior.

**Pull request merged**